code/modules/mob/living/simple_animal/heretic_monsters.dm

```diff
@@ -0,0 +1,436 @@
+ */
+/mob/living/simple_animal/hostile/heretic_summon/armsy/proc/heal()
+	if(back)
+		back.heal()
+		return
+
+	adjustBruteLoss(-maxHealth * 0.5, FALSE)
+	adjustFireLoss(-maxHealth * 0.5, FALSE)
+
+	if(health < maxHealth * 0.8)
+		return
+
+	if(++current_stacks < stacks_to_grow)
+		return
+
+	var/mob/living/simple_animal/hostile/heretic_summon/armsy/prev = new type(drop_location(), FALSE)
+	icon_state = "armsy_mid"
+	icon_living = "armsy_mid"
+	back = prev
+	prev.icon_state = "armsy_end"
+	prev.icon_living = "armsy_end"
+	prev.front = src
+	prev.AIStatus = AI_OFF
+	current_stacks = 0
+
+/mob/living/simple_animal/hostile/heretic_summon/armsy/Shoot(atom/targeted_atom)
+	GiveTarget(targeted_atom)
+	AttackingTarget()
+
+/mob/living/simple_animal/hostile/heretic_summon/armsy/AttackingTarget()
+	if(istype(target, /obj/item/bodypart/r_arm) || istype(target, /obj/item/bodypart/l_arm))
+		playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+		qdel(target)
+		heal()
+		return
+	if(target == back || target == front)
+		return
+	if(back)
+		back.GiveTarget(target)
+		back.AttackingTarget()
+	if(!Adjacent(target))
+		return
+	do_attack_animation(target)
+
+	if(iscarbon(target))
+		var/mob/living/carbon/carbon_target = target
+		if(HAS_TRAIT(carbon_target, TRAIT_NODISMEMBER))
+			return
+		var/list/parts_to_remove = list()
+		for(var/obj/item/bodypart/bodypart in carbon_target.bodyparts)
+			if(bodypart.body_part != HEAD && bodypart.body_part != CHEST && bodypart.body_part != LEG_LEFT && bodypart.body_part != LEG_RIGHT)
+				if(bodypart.dismemberable)
+					parts_to_remove += bodypart
+
+		if(parts_to_remove.len && prob(10))
+			var/obj/item/bodypart/lost_arm = pick(parts_to_remove)
+			lost_arm.dismember()
+
+	return ..()
+
+/mob/living/simple_animal/hostile/heretic_summon/armsy/prime
+	name = "Lord of the Night"
+	real_name = "Master of Decay"
+	maxHealth = 400
+	health = 400
+	melee_damage = 50
+
+/mob/living/simple_animal/hostile/heretic_summon/armsy/prime/Initialize(mapload, spawn_bodyparts = TRUE, worm_length = 9)
+	. = ..()
+	var/matrix/matrix_transformation = matrix()
+	matrix_transformation.Scale(1.4, 1.4)
+	transform = matrix_transformation
+
+/mob/living/simple_animal/hostile/heretic_summon/rust_spirit
+	name = "Rust Walker"
+	real_name = "Rusty"
+	desc = "An incomprehensible abomination. Everywhere it steps, it appears to be actively seeping life out of its surroundings."
+	icon_state = "rust_walker_s"
+	icon_living = "rust_walker_s"
+	status_flags = CANPUSH
+	maxHealth = 75
+	health = 75
+	melee_damage = 20
+	sight = SEE_TURFS
+	spells_to_add = list(
+		/obj/effect/proc_holder/spell/aoe_turf/rust_conversion/small,
+		/obj/effect/proc_holder/spell/targeted/projectile/dumbfire/rust_wave/short,
+	)
+
+/mob/living/simple_animal/hostile/heretic_summon/rust_spirit/setDir(newdir)
+	. = ..()
+	if(newdir == NORTH)
+		icon_state = "rust_walker_n"
+	else if(newdir == SOUTH)
+		icon_state = "rust_walker_s"
```

Comment:
why the fuck does it do this. why is it not one icon with dirs

Reply:
Looks like because it's not meant to turn left or right, only up and down. Would that be possible to replicate with just icons?

Reply:
![image](https://user-images.githubusercontent.com/110184118/226026320-7a82f17b-21af-40bd-9258-765c8a2f9a9e.png)